patches/api/0420-More-native-map-PDC-methods-use-adventure-Key.patch

```diff
@@ -0,0 +1,235 @@
++     * @throws NullPointerException     if the type is null
++     * @throws IllegalArgumentException if no suitable adapter will be found for
++     *                                  the {@link PersistentDataType#getPrimitiveType()}
++     */
++    <T, Z> @NotNull Z computeIfAbsent(@NotNull net.kyori.adventure.key.Key key, @NotNull PersistentDataType<T, Z> type,
++                                     @NotNull Function<net.kyori.adventure.key.Key, Z> mappingFunction);
++
++    /**
++     * Attempts to remap a current metadata value on your {@link PersistentDataHolder} instance if the value
++     * is currently present and the mapping function returns a non-null value
++     * The key is removed if your remapped value is null
++     *
++     * @param key               the key this value will be stored under
++     * @param type              the type this tag uses
++     * @param remappingFunction the function to compute your value
++     * @param <T>               the generic java type of the tag value
++     * @param <Z>               the generic type of the object to store
++     *
++     * @return The currently existing metadata within your {@link PersistentDataHolder} instance
++     *
++     * @throws NullPointerException     if the key is null
++     * @throws NullPointerException     if the type is null
++     * @throws IllegalArgumentException if no suitable adapter will be found for
++     *                                  the {@link PersistentDataType#getPrimitiveType()}
++     */
++    <T, Z> @Nullable Z computeIfPresent(@NotNull net.kyori.adventure.key.Key key, @NotNull PersistentDataType<T, Z> type,
++                                      @NotNull BiFunction<net.kyori.adventure.key.Key, Z, Z> remappingFunction);
++
++    /**
++     * Attempts to remap an {@link PersistentDataHolder} metadata value using a supplied mapping function
++     * Your key is removed if your remapped value is null
++     *
++     * @param key               the key this value will be stored under
++     * @param type              the type this tag uses
++     * @param remappingFunction the function to compute your value
++     * @param <T>               the generic java type of the tag value
++     * @param <Z>               the generic type of the object to store
++     *
++     * @return The currently existing metadata within your {@link PersistentDataHolder} instance
++     *
++     * @throws NullPointerException     if the key is null
++     * @throws NullPointerException     if the type is null
++     * @throws IllegalArgumentException if no suitable adapter will be found for
++     *                                  the {@link PersistentDataType#getPrimitiveType()}
++     */
++    <T, Z> @Nullable Z compute(@NotNull net.kyori.adventure.key.Key key, @NotNull PersistentDataType<T, Z> type,
++                             @NotNull BiFunction<net.kyori.adventure.key.Key, Z, Z> remappingFunction);
```

Comment:
The `remappingFunction` type should be `BiFunction<? super Key, ? super Z, ? extends Z>`.